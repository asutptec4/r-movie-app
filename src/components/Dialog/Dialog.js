import PropTypes from 'prop-types';
import React, { useEffect } from 'react';
import ReactDOM from 'react-dom';

import { defaultHandler } from '../../utils/util-func';

import './Dialog.scss';

const modalRoot = document.getElementById('modal-root');
const toggleScroll = () => {
  document.querySelector('html').classList.toggle('scroll-lock');
};

const Dialog = ({ children, handleClose }) => {
  useEffect(() => {
    toggleScroll();
    return () => {
      toggleScroll();
    };
  });

  return (
    <>
      {ReactDOM.createPortal(
        <div className="dialog">
          <div className="dialog-body">
            <button className="dialog-close" onClick={handleClose}>
              <svg className="dialog-close-icon" viewBox="0 0 40 40">
                <path d="M 10,10 L 30,30 M 30,10 L 10,30" />
              </svg>
            </button>
            <div className="dialog-content">{children}</div>
          </div>
        </div>,
        modalRoot,
      )}
    </>
  );
};

Dialog.defaultProps = {
  handleClose: defaultHandler,
};

Dialog.propTypes = {
  children: PropTypes.node.isRequired,
  handleClose: PropTypes.func,
};

export default Dialog;
