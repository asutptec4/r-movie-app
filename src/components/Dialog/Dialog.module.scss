@import '../../scss/index.scss';

.dialog {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: 10;
  transform: translateZ(0);
  background-color: rgba(0, 0, 0, 0.8);
  backdrop-filter: blur(8px);
  overflow-y: auto;

  .dialogBody {
    position: fixed;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    background-color: $content-bg-color;
  }
  
  @media screen and (min-width: $media-width) {
    .dialogBody {
      left: 50%;
      top: 10%;
      height: auto;
      padding: 2em 1em;
      transform: translate(-50%, 0);
      max-width: 30em;
    }
  }

  .dialogClose {
    position: absolute;
    top: 0;
    right: 0;
    padding: 0.5em;
    line-height: 1;
    background: $content-bg-color;
    border: 0;
    box-shadow: 0;
    cursor: pointer;
  }

  .dialogCloseIcon {
    width: 40px;
    height: 40px;
    fill: transparent;
    stroke: $label-color;
    stroke-linecap: round;
    stroke-width: 2;
  }

  .dialogContent {
    padding: 0em 0.5em;
  }
}
