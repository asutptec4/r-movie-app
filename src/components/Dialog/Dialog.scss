@import '../../scss/index.scss';

.dialog {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: 10;
  transform: translateZ(0);
  background-color: rgba(0, 0, 0, 0.8);
  backdrop-filter: blur(8px);
  overflow-y: auto;

  .dialog-body {
    position: fixed;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    background-color: $content-bg-color;
  }
  
  @media screen and (min-width: $media-width) {
    .dialog-body {
      left: 50%;
      top: 10%;
      height: auto;
      padding: 2em 1em;
      transform: translate(-50%, 0);
      max-width: 30em;
    }
  }

  .dialog-close {
    position: absolute;
    top: 0;
    right: 0;
    padding: 0.5em;
    line-height: 1;
    background: $content-bg-color;
    border: 0;
    box-shadow: 0;
    cursor: pointer;
  }

  .dialog-close-icon {
    width: 40px;
    height: 40px;
    fill: transparent;
    stroke: $label-color;
    stroke-linecap: round;
    stroke-width: 2;
  }

  .dialog-content {
    padding: 0em 0.5em;
  }
}
