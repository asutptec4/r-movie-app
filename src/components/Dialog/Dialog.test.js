import { render } from '@testing-library/react';
import React from 'react';

import Dialog from './Dialog';

const TestChildComponent = () => <div>Child component</div>;

describe('<Dialog />', () => {
  test('should render children', () => {
    const { container } = render(
      <Dialog handleClose={jest.fn()}>
        <TestChildComponent />,
      </Dialog>,
    );
    expect(container).toMatchSnapshot();
  });
});
