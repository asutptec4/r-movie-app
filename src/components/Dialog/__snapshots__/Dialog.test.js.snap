// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dialog /> should render children 1`] = `<div />`;
